File: benchmarks/beginner/B1-fibonacci/javascript/main.js
function fib(n) {
  if (n <= 1) return n;
  return fib(n - 1) + fib(n - 2);
}

function main() {
  const N = 40;
  const RUNS = 10;
  const timings = [];

  console.log("=== BENCHMARK: B1 - Fibonacci ===");
  console.log("Language: JavaScript");
  console.log(`Runtime:  Node ${process.version}`);
  console.log();

  for (let i = 1; i <= RUNS; i++) {
    const start = process.hrtime.bigint();
    const result = fib(N);
    const elapsed = Number(process.hrtime.bigint() - start) / 1e6; // ms
    timings.push({ elapsed, result });
    console.log(`Run ${String(i).padStart(2)}: ${elapsed.toFixed(2)} ms`);
  }

  console.log();
  const times = timings.map((t) => t.elapsed);
  const min = Math.min(...times);
  const avg = times.reduce((a, b) => a + b, 0) / times.length;
  const max = Math.max(...times);
  const variance = times.reduce((a, t) => a + (t - avg) ** 2, 0) / times.length;
  const stddev = Math.sqrt(variance);

  console.log(`Min:    ${min.toFixed(2)} ms`);
  console.log(`Avg:    ${avg.toFixed(2)} ms`);
  console.log(`Max:    ${max.toFixed(2)} ms`);
  console.log(`StdDev: ${stddev.toFixed(2)} ms`);
  console.log();
  const result = timings[timings.length - 1].result;
  const status = result === 102334155 ? "✓" : "✗";
  console.log(`Verification: ${result} ${status}`);
}

main();

File: benchmarks/beginner/B1-fibonacci/rust/main.rs
use std::hint::black_box;
use std::time::Instant;

#[inline(never)]
fn fib(n: u32) -> u64 {
    if n <= 1 {
        return n as u64;
    }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    const N: u32 = 40;
    const RUNS: usize = 10;
    let mut timings = [0.0f64; RUNS];

    println!("=== BENCHMARK: B1 - Fibonacci ===");
    println!("Language: Rust");
    println!("Runtime:  rustc (native)");
    println!();

    let mut result = 0u64;
    for i in 0..RUNS {
        let start = Instant::now();
        result = black_box(fib(black_box(N)));
        let elapsed = start.elapsed().as_secs_f64() * 1000.0; // ms
        timings[i] = elapsed;
        println!("Run {:2}: {:.2} ms", i + 1, elapsed);
    }

    println!();
    let min = timings.iter().cloned().fold(f64::INFINITY, f64::min);
    let max = timings.iter().cloned().fold(f64::NEG_INFINITY, f64::max);
    let avg = timings.iter().sum::<f64>() / RUNS as f64;
    let variance = timings.iter().map(|t| (t - avg).powi(2)).sum::<f64>() / RUNS as f64;
    let stddev = variance.sqrt();

    println!("Min:    {:.2} ms", min);
    println!("Avg:    {:.2} ms", avg);
    println!("Max:    {:.2} ms", max);
    println!("StdDev: {:.2} ms", stddev);
    println!();
    let status = if result == 102334155 { "✓" } else { "✗" };
    println!("Verification: {} {}", result, status);
}
